Sources/Penny/Services/CacheService/DefaultCacheService.swift

```diff
@@ -0,0 +1,51 @@
+import Logging
+import AsyncHTTPClient
+import SotoS3
+
+actor DefaultCachesService: CachesService {
+
+    var cacheRepo: S3CacheRepository!
+    let logger = Logger(label: "DefaultCachesService")
+
+    private init() { }
+
+    static let shared = DefaultCachesService()
+
+    func initialize(httpClient: HTTPClient) {
+        let awsClient = AWSClient(httpClientProvider: .shared(httpClient))
+        self.cacheRepo = .init(awsClient: awsClient, logger: self.logger)
+    }
```

Comment:
Ouch. Please dear Gods let [SE-0327](https://github.com/apple/swift-evolution/blob/main/proposals/0327-actor-initializers.md) actually show up in a Swift release soon...